src/Engine/GraphLayout/GraphLayout.cs

```diff
@@ -108,6 +108,9 @@ public void AddNode(Guid guid, double width, double height, double x, double y,
         /// <param name="endY">The y coordinate of the connector's right end point.</param>
         public void AddEdge(Guid startId, Guid endId, double startX, double startY, double endX, double endY)
         {
+            //Validates that the two nodes that will be used to create the Edge exist
+            if (Nodes.Where(node => node.Id == startId).Count() == 0  || Nodes.Where(node => node.Id == endId).Count() == 0) return;
```

Comment:
Maybe use !Any() instead?

Reply:
updated in the next commit: https://github.com/DynamoDS/Dynamo/pull/15495/commits/0f97fe1be9df374243be1cbfe38238364f7bce45